### PR TITLE
ci: use github token permissions instead of personal access tokens

### PR DESCRIPTION
**Pull request by nunofgs**
### Description

Replaces PATs currently being used for various CI actions with `GITHUB_TOKEN` and respective [permissions](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/).

These changes match the work being done on https://github.com/customerio/customerio-ios/pull/261.

### Note to reviewers
<details>
  <summary>The `deploy-maven-central` and `deploy-git-tag` workflows were merged into a single workflow.</summary>
  <br/>
  <ul>
   <li>This was required because events generated by the use of `GITHUB_TOKEN` <a href="https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow">will not trigger</a> other workflows.</li>
  <li>
  This will be the new flow:
  Screenshot 2023-02-10 at 12 06 46
  </li>
</details>

<details>
  <summary>The `action-semantic-release-justwork` step was removed.</summary>
  <br/>
  <ul>
   <li>This is no longer necessary since we're not updating long-running tags anymore. (such as `v1`, `v2`, etc.)</li>
  </ul>
</details>

<details>
  <summary>Pull requests are no longer created as `develop` branch is being phased out.</summary>
  <br/>
  <ul>
   <li>Previously this was only done if the automatic merge failed.</li>
  </ul>
</details>

<details>
  <summary>actions/checkout was bumped from v2 to v3.</summary>
  <br/>
  <ul>
   <li>Just cleanup!</li>
  </ul>
</details>

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-android/develop/docs/dev-notes/GIT-WORKFLOW.md).
- [x] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request.
- [ ] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1).
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request and it will automatically be merged.

### CODE REVIEW

File: .github/workflows/deploy-sdk.yml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        env:
+          # Incoming webhook URL that sends message into the correct Slack channel.
+          # Help on how to get the webhook URL: https://github.com/marketplace/actions/slack-send#setup-2
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_NOTIFY_RELEASES_WEBHOOK_URL }}
+          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+
+  deploy-sonatype:
+    name: Deploy SDK to Maven Central
+    needs: [deploy-git-tag]
+    if: needs.deploy-git-tag.outputs.new_release_published == 'true' # only run if a git tag was made.
```
Comment (levibostian) [inline .github/workflows/deploy-sdk.yml:122]: the `deploy-git-tag` job needs to have `outputs` setup in order for this to work.

File: .github/workflows/deploy-sdk.yml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 11
+      - name: Push to Sonatype servers
+        run: MODULE_VERSION=${{ github.event.release.tag_name }} ./scripts/deploy-code.sh
```
Comment (levibostian) [inline .github/workflows/deploy-sdk.yml:132]:
```suggestion
        run: MODULE_VERSION=${{ needs.deploy-git-tag.outputs.new_release_version }} ./scripts/deploy-code.sh
```

`github.event.release` will be null now that we have combined the 2 CI workflow files into 1.

File: .github/workflows/deploy-sdk.yml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+# Only run after a pull request has been merged. This is because
+# bot account runs write operations on the github repo to push a tag.
+on:
+  push:
+    branches: [main] # all branches where deployments currently occur. Make sure this list matches list of branches in  `.releaserc` file.
+
+permissions:
+  contents: write # access to push the git tag
+  issues: write # Bot creates an issue if there is an issue during deployment process
```
Comment (levibostian) [inline .github/workflows/deploy-sdk.yml:11]:
```suggestion
  issues: write # Bot creates an issue if there is an issue during deployment process
  pull-requests: write # allow bot to make comments on PRs after they get deployed
```

Comment (nunofgs) [reply .github/workflows/deploy-sdk.yml:11]: @levibostian we no longer use the `levibostian/action-sync-branches` which would create a PR (and required these permissions).  As far as I can see semantic-release also doesn't create a comment on the PR.

Do you still feel we need this?

Comment (levibostian) [reply .github/workflows/deploy-sdk.yml:11]: semantic-release [makes comments on PRs](https://github.com/customerio/customerio-ios/pull/230#issuecomment-1339556047). But, github now copies this feature:

CleanShot 2023-02-13 at 12 07 33@2x

....so, I suppose we don't need this feature anymore.

Comment (nunofgs) [reply .github/workflows/deploy-sdk.yml:11]: Ah, I see. It's the `semantic-release/github` plugin. They also mention there's a `failComment` that may be worth keeping: https://github.com/semantic-release/github#failcomment

Shall I re-add?

**Review by levibostian — APPROVED** [review]

**Pull request merged**